tests/unit/CompareRaftMembers/KeyValueStreamDiff.js

```diff
@@ -0,0 +1,309 @@
+const async = require('async');
+const stream = require('stream');
+
+const KeyValueStreamDiff = require('../../../CompareRaftMembers/KeyValueStreamDiff');
+
+class MockKeyValueStream extends stream.Readable {
+    constructor(listingToSend) {
+        super({ objectMode: true });
+
+        this.listingToSend = listingToSend;
+    }
+
+    _read() {
+        process.nextTick(() => {
+            if (this.listingToSend.length === 0) {
+                this.push(null);
+            } else {
+                const item = this.listingToSend.shift();
+                this.push(item);
+            }
+        });
+    }
+}
+
+describe('KeyValueStreamDiff', () => {
+    describe('should output differences between streams of { key, value } items', () => {
+        [
+            {
+                desc: 'with empty streams',
+                streamEntries: [
+                    [],
+                    [],
+                ],
+                expectedOutput: [],
+            },
+            {
+                desc: 'with respectively zero and one entry in each stream',
+                streamEntries: [
+                    [],
+                    [
+                        { key: 'bucket/key1', value: '{}' },
+                    ],
+                ],
+                expectedOutput: [
+                    [
+                        null,
+                        { key: 'bucket/key1', value: '{}' },
+                    ],
+                ],
+            },
+            {
+                desc: 'with respectively one and zero entry in each stream',
```

Comment:
could you add a note why this is not symmetric w.r.t. to the above case and needs to be tested seperately? Or is this intended as a sanity check asserting streams params can be passed in either order?

Reply:
it's a sanity check, the function logic is symmetric although the result itself is different in each case (but symmetric). I don't think it's super useful to add a comment about this but I don't mind adding it